bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/projects/JavaProjectInfo.java

```diff
@@ -403,6 +404,16 @@ private Entry toJavaSourceFileOrLabelEntry(String srcFileOrLabel) throws CoreExc
 
         // treat as label if it looks like one
         if (shouldTreatAsLabel(srcFileOrLabel)) {
+            var target = bazelPackage.getBazelTarget(srcFileOrLabel);
+            if ((target != null) && "filegroup".equals(target.getRuleClass())) {
+                var attributes = target.getRuleAttributes();
+                var srcs = attributes.getStringList(SRCS);
+                if (srcs != null) {
+                    for (String src : srcs) {
+                        return toJavaSourceFileOrLabelEntry(src);
+                    }
+                }
+            }
```

Comment:
I like the idea of supporting `filegroup` within the same package. This should be doable. But we need to make it proper and figure out a way to add all possible sources that way. Just one might not be sufficient.

Reply:
Do you have any example of such a source?

Reply:
When the `filegroup` is a `glob` matching to multiple `.java` files. Bazel should give multiple entries as input in response.

Reply:
Fixed.